### PR TITLE
use fingerprint and password at same time

### PR DESCRIPTION
**Pull request by Sasasu**
previously on #1038 

Fingerprintlogin works like Autologin, the logic is:

1. call `fingerprintLogin()` after `sddm-greeter` started
2. when user use an empty password to login switch to fingerprint login

this PR need `sddm.conf` change, see `sddm.conf.rst.in`

### CODE REVIEW

**Comment by MatthewZMD** [comment]
Any updates?

**Comment by Sasasu** [comment]
usable

**Comment by shivindera** [comment]
bump!

**Comment by MatthewZMD** [comment]
@plfiorini can we get some updates as to this patch? can it be merged?

**Comment by skwidge** [comment]
Can this please be merged?

**Comment by WMP** [comment]
Hello, i want to try this, how can i register my fingerprint?

**Comment by Sasasu** [comment]
@WMP 
Archlinux Wiki has a detailed tutorial about how to setup fingerprint login.

https://wiki.archlinux.org/index.php/Fprint

but you should completely ignore the section about pam. this patch includes a pam file, which is enough to make SDDM login using fingerprint.

TLDR:
1. install `fprint` and make sure `fprintd` is running
2. use `fprintd-enroll` to enroll one finger.
3. use `fprintd-verify` to check if it is working fine
4. apply this patch, modify `/etc/sddm.conf` and enjoy

**Comment by MatthewZMD** [comment]
Can this patch really be merged?

**Comment by carlosdem** [comment]
This patch works well for me on neon 20.04, as long as I copy the contents of /etc/pam.d/gdm-fingerprint to /etc/pam.d/sddm-fingerprint.  Really just a distro packaging issue around the usage of selinux as far as I can tell.

**Comment by tynsh** [comment]
As someone not familiar with the codebase and only using sddm as a user, here are some suggestions:

Is the change in sddm.conf really necessary? During login, the sddm gui already shows a list of users and preselects one. It would be nice, if sddm would try to receive a fingerprint for the selected user and retry with an other as soon as the user selects a different username. As for users without enrolled fingerprints, fprintd responds accordingly.

I'm not sure if the lockscreen is part of sddm, but in multi-seat environments this could be a potential issue due to multiple lockscreens being active. But this could be partly resolved by only polling for a fingerprint permanently, if only one session is active and in case of multiple sessions only polling for a couple of seconds after the last (mouse or keyboard) input.

Just my 2 cents :)

**Comment by Sasasu** [comment]
> Is the change in sddm.conf really necessary? During login, the sddm gui already shows a list of users and preselects one.

No, but for this implement it is necessary. 

for now, I think the background server process can't get the username which current display on the frontend.

`Windows Hello` or `iOS Face ID` has a very good user experience, wish SDDM also can do this.

there are some PR try to implement those frameworks https://github.com/sddm/sddm/pull/802 https://github.com/sddm/sddm/issues/781

>  if the lockscreen is part of sddm.

lockscreen is not a part of SDDM.  Usually, the lock screen is part of the desktop environment (DE), not the desktop manager(DM).

In fact, lockscreen is a screensaver. very interesting.

**Comment by zeertzjq** [comment]
On my laptop with Arch Linux it seems `fprintd.service` is started later than `sddm.service` (maybe after I log in with password) so at the first SDDM login after boot fingerprint login always fails. After adding `Wants=fprintd.service` to `/usr/lib/systemd/system/sddm.service` fingerprint login works after pressing `Enter` at SDDM login screen.

**Comment by espidev** [comment]
Any update on this being merged?

**Comment by MatthewZMD** [comment]
@plfiorini @davidedmundson Tbh I'm disappointed by the lack of feedback from the sddm developers.

Seriously, this patch is from 2019 and confirmed usable in Jan 2020. What's taking so long to merge or, at the very least, comment on why this cannot be merged?

**Comment by davama** [comment]
is this still usable?
I would like to give it a try but it says "out-of-date"...so not sure
Thanks!

**Comment by AlvinZhu** [comment]
This patch does not work. Login always failed.
I applied this patch on arch linux's package sddm 0.18.1-3.

**Comment by davama** [comment]
@AlvinZhu 
Thank you for the update.
That stinks :disappointed:

**Comment by Sasasu** [comment]
> is this still usable?
> I would like to give it a try but it says "out-of-date"...so not sure
> Thanks!

“out of date” means this branch is older than current develop branch. You can apply the patch to current develop branch by yourself.


> This patch does not work. Login always failed.
> I applied this patch on arch linux's package sddm 0.18.1-3.

Can you check your fprintd settings use fprintd-verif?

**Comment by AlvinZhu** [comment]
> > is this still usable?
> > I would like to give it a try but it says "out-of-date"...so not sure
> > Thanks!
> 
> “out of date” means this branch is older than current develop branch. You can apply the patch to current develop branch by yourself.
> 
> > This patch does not work. Login always failed.
> > I applied this patch on arch linux's package sddm 0.18.1-3.
> 
> Can you check your fprintd settings use fprintd-verif?

fprintd-verify match
I checked.

**Comment by AlvinZhu** [comment]
> > is this still usable?
> > I would like to give it a try but it says "out-of-date"...so not sure
> > Thanks!
> 
> “out of date” means this branch is older than current develop branch. You can apply the patch to current develop branch by yourself.
> 
> > This patch does not work. Login always failed.
> > I applied this patch on arch linux's package sddm 0.18.1-3.
> 
> Can you check your fprintd settings use fprintd-verif?

The log of sddm:
```
Sep 28 21:39:13 Alvin-Laptop sddm[1111]: starting sddm-helper with ("--socket", "/tmp/sddm-authxxxxxx", "--id", "1", "--start", "/usr/bin/startplasma-x11", "--user", "alvin", "--fingerprintlogin")
Sep 28 21:39:13 Alvin-Laptop sddm-helper[2044]: [PAM] using service name "sddm-fingerprint"
Sep 28 21:39:13 Alvin-Laptop sddm-helper[2044]: PAM unable to dlopen(/usr/lib/security/pam_tally.so): /usr/lib/security/pam_tally.so: cannot open shared object file: No such file or directory
Sep 28 21:39:13 Alvin-Laptop sddm-helper[2044]: PAM adding faulty module: /usr/lib/security/pam_tally.so
Sep 28 21:39:13 Alvin-Laptop sddm-helper[2044]: [PAM] Starting...
Sep 28 21:39:13 Alvin-Laptop sddm-helper[2044]: [PAM] Authenticating...
Sep 28 21:39:13 Alvin-Laptop sddm-helper[2044]: [PAM] Preparing to converse...
Sep 28 21:39:13 Alvin-Laptop sddm-helper[2044]: [PAM] Conversation with 1 messages
Sep 28 21:39:13 Alvin-Laptop sddm[1111]: Authentication information: "Place your right index finger on the fingerprint reader"
Sep 28 21:39:14 Alvin-Laptop sddm-helper[2044]: [PAM] authenticate: Module is unknown
Sep 28 21:39:14 Alvin-Laptop sddm-helper[2044]: [PAM] returning.
Sep 28 21:39:14 Alvin-Laptop sddm[1111]: Authentication error: "Module is unknown"
Sep 28 21:39:14 Alvin-Laptop sddm-helper[2044]: [PAM] Ended.
Sep 28 21:39:14 Alvin-Laptop sddm[1111]: Auth: sddm-helper exited with 1
```

```
% ls /usr/lib/security
pam_access.so     pam_debug.so  pam_exec.so       pam_filter.so         pam_group.so     pam_lastlog.so    pam_loginuid.so   pam_namespace.so  pam_pwhistory.so  pam_securetty.so  pam_stress.so        pam_time.so       pam_unix.so      pam_winbind.so
pam_apparmor.so   pam_deny.so   pam_faildelay.so  pam_fprintd.so        pam_issue.so     pam_limits.so     pam_mail.so       pam_nologin.so    pam_pwquality.so  pam_setquota.so   pam_succeed_if.so    pam_timestamp.so  pam_usertype.so  pam_xauth.so
pam_cap.so        pam_echo.so   pam_faillock.so   pam_ftp.so            pam_keyinit.so   pam_listfile.so   pam_mkhomedir.so  pam_oath.so       pam_rhosts.so     pam_shells.so     pam_systemd_home.so  pam_tty_audit.so  pam_warn.so
pam_cifscreds.so  pam_env.so    pam_filter        pam_gnome_keyring.so  pam_kwallet5.so  pam_localuser.so  pam_motd.so       pam_permit.so     pam_rootok.so     pam_snapper.so    pam_systemd.so       pam_umask.so      pam_wheel.so
```

```
% yay -Qo /usr/lib/security
/usr/lib/security/ is owned by apparmor 2.13.4-6
/usr/lib/security/ is owned by cifs-utils 6.11-1
/usr/lib/security/ is owned by fprintd 1.90.1-1
/usr/lib/security/ is owned by gnome-keyring 1:3.36.0-1
/usr/lib/security/ is owned by kwallet-pam 5.19.5-1
/usr/lib/security/ is owned by libcap 2.43-1
/usr/lib/security/ is owned by libpwquality 1.4.2-3
/usr/lib/security/ is owned by oath-toolkit 2.6.2-8
/usr/lib/security/ is owned by pam 1.4.0-3
/usr/lib/security/ is owned by samba 4.12.6-1
/usr/lib/security/ is owned by snapper 0.8.14-1
/usr/lib/security/ is owned by systemd 246.6-1
```

```
% uname -r
5.8.12-arch1-1
```

**Comment by Sasasu** [comment]
looks pam 1.4 use `pam_faillock.so` to replace `pam_tally.so`.

apply this patch on `/etc/pam.d/sddm-autologin`

```patch
-auth        required    pam_tally.so file=/var/log/faillog onerr=succeed
+auth        required    pam_faillock.so preauth
```

**Comment by AlvinZhu** [comment]
> looks pam 1.4 use `pam_faillock.so` to replace `pam_tally.so`.
> 
> apply this patch on `/etc/pam.d/sddm-fingerprint`
> 
> ```diff
> -auth        required    pam_tally.so file=/var/log/faillog onerr=succeed
> +auth        required    pam_faillock.so preauth
> ```
> 
> and you can debug pam using `pamtester`
> 
> ```shell
> sudo pamtester -v sddm-fingerprint alvin authenticate
> ```
> 
> I will update this PR after this code change
> https://github.com/sddm/sddm/blob/9708eecc2cbcb988ce520126e7280412a4c59d44/services/sddm-autologin.pam#L3

It works. thank you.
Btw,  is there a patch for lockscreen, or any plans?

**Comment by Sasasu** [comment]
>  is there a patch for lockscreen, or any plans?

no, lock screen is a part of KDE.

File: src/helper/Backend.h [hunk 1]
```diff
@@ -36,6 +36,7 @@ namespace SDDM {
         static Backend *get(HelperApp *parent);
 
         void setAutologin(bool on = true);
+        void setFingerPtintLogin(bool on = true);
```
Comment (oxc) [inline src/helper/Backend.h]: typo. should be `setFingerPrintLogin`.

Comment (Sasasu) [inline src/helper/Backend.h:39]:
```suggestion
        void setFingerPrintLogin(bool on = true);
```

Comment (Sasasu) [reply src/helper/Backend.h]: thanks

**Comment by shivindera** [comment]
How can I patch this to an existing installation of sddm? Can someone help?

**Comment by Sasasu** [comment]
> How can I patch this to an existing installation of sddm?

you can not. this patch needs re-compile sddm.

but you can get everything you will need from the official archlinux build script.

https://github.com/archlinux/svntogit-packages/blob/packages/sddm/trunk/PKGBUILD

**Comment by SwiftCoderJoe** [comment]
Bump. This patch has been sitting in limbo for over a year... can we get this reviewed?

**Comment by Sasasu** [comment]
push -f and resolve conflict

**Comment by Pointedstick** [comment]
Ping @plfiorini we would really like this as it is needed to support a related feature in KDE Plasma. :)

**Comment by zomp** [comment]
@Pointedstick As many people are the only users of their computers, they may not need SDDM support, so it might get deprioritized – for these people may be preferable SDDM "auto-login" followed by lockscreen right away – once they reach the computer after its start, everything runs just after logging, because there is no need to wait for Plasma to start.

Users requiring SDDM would still need its support, but KDE Plasma could enable the support for the remaining users who are fine with lockscreen alone.

**Comment by Orhideous** [comment]
Any chance to get this merged?

**Comment by Thelm76** [comment]
I hope this will be merged in a near future :D I think there is a PR waiting for it in KDE Plasma's repo to improve fingerprint support!

**Comment by PlaidPhantom** [comment]
It would be great to see this approved, particularly with the release of the Framework Laptop. They made sure to choose a Linux-supported fingerprint reader, so it's disappointing for it to still be the one component of my laptop that I cannot use seamlessly with KDE.

**Comment by eiis1000** [comment]
Can this please be merged?

**Comment by mkblast** [comment]
Is there any reason to not merge this?

**Comment by Pointedstick** [comment]
Well for one, there are merge conflicts. :) But yes, the lack of reviews is a bit dispiriting. Hopefully @plfiorini or @davidedmundson will be able to at some point.

**Comment by Sasasu** [comment]
~~opps. I have no idea why I close this PR.~~
but there are no conflicts anymore.

File: src/auth/Auth.cpp [hunk 2]
```diff
@@ -226,12 +227,13 @@ namespace SDDM {
             Q_EMIT qobject_cast<Auth*>(parent())->error(child->errorString(), ERROR_INTERNAL);
         }
 
-        if (exitCode == HELPER_SUCCESS)
+        if (exitCode == HELPER_SUCCESS) {
             qDebug() << "Auth: sddm-helper exited successfully";
+            emit qobject_cast<Auth*>(parent())->finished(static_cast<Auth::HelperExitStatus>(exitCode));
```
Comment (CarlSchwan) [inline src/auth/Auth.cpp:231]:
```suggestion
            Q_EMIT qobject_cast<Auth*>(parent())->finished(static_cast<Auth::HelperExitStatus>(exitCode));
```

File: src/auth/Auth.cpp [hunk 3]
```diff
@@ -338,6 +344,12 @@ namespace SDDM {
         }
     }
 
+    void Auth::setFingerprintlogin(bool on){
+        if(on != d->fingerprintlogin){
+            d->fingerprintlogin = on;
+        }
```
Comment (CarlSchwan) [inline src/auth/Auth.cpp:346]:
```suggestion
        }
        Q_EMIT fingerprintloginChanged();
```

**Review by CarlSchwan — COMMENTED** [review]
Just some small code comments. Didn't test it as I don't have a fingerprint sensor

**Comment by 0x7CFE** [comment]
Any idea when this can be merged or at least reviewed? I would really like to use a fingerprint sensor on my Framework laptop. It works awesome and it's ridiculous to know that it can't be used because some patch is ready but hanging without motion for two freaking years :facepalm: 

Kudos to @Sasasu for keeping it up to date :heart:

**Comment by CarlSchwan** [comment]
Maybe @aleixpol or @davidedmundson can take a look :) I looked at the code 2 weeks ago and it looks fine, but I can't test it as I don't have a fingerprint reader

**Comment by SARankDirector-Minecraft** [comment]
bump

**Comment by Emojigit** [comment]
This is the most long-lasting PR I've ever seen in the ENTIRE coding community. As it is usable and useful, please merge it (before Ubuntu 22.04 feature freeze, whenever possible)

**Comment by jannis3005** [comment]
I'd have the hardware to test it, but I've never done anything with sddm development / building. I'll see if I can find time over the weekend to figure out how to build this and test it.

**Comment by Leptopoda** [comment]
As this has gone stale again I'd like to ask what can be done to bring this forward.

Can we just test this?

**Comment by Orhideous** [comment]
Suddenly I have a spare notebook with fingerprint sensor. I would greatly appreciate any notes/scenarios/test plan for this feature :)

**Comment by madsl** [comment]
Why does the patch add "--fingerprintlogin" to the argument list for the session you log in to (e.g. ? It makes KDE Plasma Wayland session 5.26 crash...
`
sddm[1851]: starting sddm-helper with ("--socket", "/tmp/sddm-authb7cc65c7-7642-4911-ba3d-079241cd9404", "--id", "1", "--start", "/usr/lib64/libexec/plasma-dbus-run-session-if-needed /usr/bin/startplasma-wayland", "--user", "mads", "--fingerprintlogin")`

Is it really necessary...?

Edit: I misread the output, sorry.

But, if I add
[Fingerprintlogin]
User=user
to sddm.conf and try to login with a Wayland session, the login process crashes...

**Comment by madsl** [comment]
It works if you add Session=plasmawayland.desktop to the config file.

**Comment by rottenpants466** [comment]
Any updates on this merge request?

**Comment by cybik** [comment]
Also interested in seeing this merged.

**Comment by IslandNolan** [comment]
bump

**Comment by marie-bnl** [comment]
It works like a charm on my side too.

Btw, for impatient Arch users, I created an AUR package: [sddm-sasasu-git](https://aur.archlinux.org/packages/sddm-sasasu-git)

**Comment by 0verEngineer** [comment]
Please review and merge this, this holds back KDEs Fingerprint feature from fully working.

**Comment by Vogtinator** [comment]
Let's get 0.20 ready and released before adding even more features.

**Comment by 0verEngineer** [comment]
> Let's get 0.20 ready and released before adding even more features.

Sounds great if this feature would actually be merged after 0.20...

**Comment by orazdow** [comment]
This would be very useful, why are they still not merging it?

**Comment by Leptopoda** [comment]
> Let's get 0.20 ready and released before adding even more features.

It's literally just two posts above. The last release was three years ago and they first want to get all the maintenance backlog done before adding new features.

**Comment by orazdow** [comment]
> > Let's get 0.20 ready and released before adding even more features.
> 
> It's literally just two posts above. The last release was three years ago and they first want to get all the maintenance backlog done before adding new features.

This would sense for a recent pr, but there's 3 years of posts above refusing to review it.

**Comment by zgana** [comment]
> Let's get 0.20 ready and released before adding even more features.

@Vogtinator  Is this going to move forward now that 0.20.0 is out?

**Comment by adonespitogo** [comment]
Hope this can be merged soon

**Comment by Sasasu** [comment]
rebase with sddm/develop. format code and remove `pam_tally.so` in the pam file.

the old commit hash is https://github.com/sddm/sddm/pull/1220/commits/36e1594ea5f1a220b881df9d156dca54092e87e3

File: data/man/sddm.conf.rst.in [hunk 4]
```diff
@@ -246,6 +246,18 @@ The `UserAuthFile=` option was removed, the file is always created as
...
 
+[Fingerprintlogin] section:
+
+`User=`
+        Name of the user who uses fingerprint login when the
+        system starts first time.
+        Default value is empty.
+
+`Session=`
+        Name of the user who uses fingerprint login when the
```
Comment (nullie) [inline data/man/sddm.conf.rst.in]: Wrong description here

**Comment by bkmo** [comment]
This builds and functions ok. I have to hit enter to get the fingerprint reader to activate. If I add the [Fingerprintlogin] section to sddm.conf it will activate the reader automatically (no password option is available this way).

**Comment by jclsn** [comment]
Would be nice if it worked without enter actually. Like starting the fingerprint reader when the SDDM starts and turning it off once you are logged in. Is that not possible?

**Comment by bkmo** [comment]
> Would be nice if it worked without enter actually. Like starting the fingerprint reader when the SDDM starts and turning it off once you are logged in. Is that not possible?

It is possible, and this PR does just that if you add the [Fingerprintlogin] section to sddm.conf. Or create a dropin file adding the section.  You will go directly to the SDDM greeter screen and the fingerprint reader will be waiting for a scan. Without adding this section the behavior is password entry or press enter to activate the scanner.

File: data/man/sddm.conf.rst.in (hunk 4 above)
Comment (z2rec) [reply data/man/sddm.conf.rst.in]: Are we just waiting for the Session description to be changed?

Comment (bkmo) [reply data/man/sddm.conf.rst.in]: Looks that way.

**Comment by Vistaus** [comment]
Looks like this won't get merged even now that 0.20 release is done. Let me guess: next response will be “Let's get 0.21 ready and released before adding even more features.”?

Even though this PR is already old *and* fingerprint readers are used everywhere these days, making it a very necessary thing to support.

**Comment by jclsn** [comment]
I also don't get it. If it works, ship it!

**Comment by Vogtinator** [comment]
IMO this needs to be designed and implemented differently. Treating fingerprint login like autologin doesn't really make sense, this needs to work OOTB without any configuration of user and session.

Themes need to be able to tell SDDM to start a fingerprint PAM auth process in the background and get feedback about progress,  in parallel to password authentication. Plasma's kscreenlocker can serve as inspiration, though the integration with SDDM's PAM code is obviously different.

**Comment by 0verEngineer** [comment]
> IMO this needs to be designed and implemented differently. Treating fingerprint login like autologin doesn't really make sense, this needs to work OOTB without any configuration of user and session.
> 
> Themes need to be able to tell SDDM to start a fingerprint PAM auth process in the background and get feedback about progress,  in parallel to password authentication. Plasma's kscreenlocker can serve as inspiration, though the integration with SDDM's PAM code is obviously different.

That sounds reasonable, however this code is done and works, should be merged even if it is not a good / the best solution because ATM IT DOES NOT WORK AT ALL.

If you think this should be implemented in another way then nothing is stopping you from doing so, this code here can be replaced in the future with the better solution, but having no solution at all is bullshit.

**Comment by Vistaus** [comment]
> > IMO this needs to be designed and implemented differently. Treating fingerprint login like autologin doesn't really make sense, this needs to work OOTB without any configuration of user and session.
> > Themes need to be able to tell SDDM to start a fingerprint PAM auth process in the background and get feedback about progress,  in parallel to password authentication. Plasma's kscreenlocker can serve as inspiration, though the integration with SDDM's PAM code is obviously different.
> 
> That sounds reasonable, however this code is done and works, should be merged even if it is not a good / the best solution because ATM IT DOES NOT WORK AT ALL.
> 
> If you think this should be implemented in another way then nothing is stopping you from doing so, this code here can be replaced in the future with the better solution, but having no solution at all is bullshit.

Exactly. The current implementation may not be the best, but has been proven to work fine. So I'd say: merge and ship it for the time being and improve the implementation over time.

**Comment by Sasasu** [comment]
> Themes need to be able to tell SDDM to start a fingerprint PAM auth process in the background and get feedback about progress.

I agree with you, but https://github.com/sddm/sddm/pull/1220#issuecomment-664087028

**Comment by Septikai** [comment]
Attempting to set this up on Arch with KDE using the AUR package, and without the `[FingerprintIogin]` section it works well and accepts a password or a fingerprint after entering a blank password, as expected after reading the comments here. With the `[Fingerprintlogin]` section (as mentioned [here](https://github.com/sddm/sddm/pull/1220#issuecomment-1745674837)), it will accept a fingerprint without the blank password, but then entering a password causes it to wait for a fingerprint, and if the fingerprint is incorrect it does not unlock regardless of the password, unless I use an incorrect fingerprint several times first, which causes it to stop accepting a fingerprint for me. Is there a way for me to set it to accept a fingerprint to unlock from the start (without entering a blank password), but still work with only password, and if so then how would I go about doing that?

**Comment by 0verEngineer** [comment]
@Vogtinator You sound like you have dived deeper into this code, i would like to implement it the best way possible but i have no experience with auth on linux. 

Can you share some tipps how you would implement this because you say above that
> Treating fingerprint login like autologin doesn't really make sense

1. How should it be treated then?
2. What should i checkout / read first?
3. Anything else i should know?

**Comment by Vogtinator** [comment]
> @Vogtinator You sound like you have dived deeper into this code, i would like to implement it the best way possible but i have no experience with auth on linux.
> 
> Can you share some tipps how you would implement this because you say above that
> 
> > Treating fingerprint login like autologin doesn't really make sense
> 
> 1. How should it be treated then?

Quoting above:

> Themes need to be able to tell SDDM to start a fingerprint PAM auth process in the background and get feedback about progress, in parallel to password authentication. Plasma's kscreenlocker can serve as inspiration, though the integration with SDDM's PAM code is obviously different.

> 2. What should i checkout / read first?

PAM documentation, maybe the code of the involved PAM auth modules. Then the kscreenlocker code which handles this pretty much as well as is currently possible.

> 3. Anything else i should know?

Every minute you spend working with PAM your sanity drops.

**Comment by jerome-cui** [comment]
When will this be merged? I hate the stupid enter on login with finger!

**Comment by Vogtinator** [comment]
> When will this be merged? I hate the stupid enter on login with finger!

See https://github.com/sddm/sddm/pull/1220#issuecomment-1829545241.

**Comment by Svenum** [comment]
I am waiting for this too...

**Comment by ZeroKun265** [comment]
I am waiting for this as well but i think it should be handled really well like kde does when the screen turns off after a while. With at least a textbox saying that it's waiting on a fingerprint input while also checking for passwords.
Sadly i wouldn't know how to help in development since i'm not an expert in this.


Edit: My solution was to setup SDDM to use autologin and then create a login script to launch the kde screen saver thing that has a better experience with fingerprint, but it also required me to set an empty password for kdewallet (which i don't even know what is used for, never really cared about it) but still, i shouldn't have to.
Besides it feels slower. It just loads the session before login instead of after but because of that it feels slower on startup. All i did was documented on the arch wiki: [Arch wiki - SDDM](https://wiki.archlinux.org/title/SDDM)

**Comment by z2rec** [comment]
> but it also required me to set an empty password for kdewallet (which i don't even know what is used for, never really cared about it) but still, i shouldn't have to.

kdewallet is a vault that stores credentials. You should **NOT** set an empty password on it.

**Comment by bkmo** [comment]
> > but it also required me to set an empty password for kdewallet (which i don't even know what is used for, never really cared about it) but still, i shouldn't have to.
> 
> kdewallet is a vault that stores credentials. You should **NOT** set an empty password on it.

You DO need an empty password if you want to open the wallet when using autologin. Another long time unsolved issue.

**Comment by ZeroKun265** [comment]
Yes I know, which is why I shouldn't be using auto login in the first place and Sddm should just offer a better experience

Edit: Also I'm not too concerned with security since I don't download a lot of software, I am very conscious about what I do on my system and nobody would try to steal my computers, I never connect to public networks etc.. so it's a good workaround for me but ideally Sddm should just offer a better option, I don't know if GDM has it fixed but I really don't wanna switch to it

**Comment by bkmo** [comment]
> Yes I know, which is why I shouldn't be using auto login in the first place and Sddm should just offer a better experience

I don't know why they don't want to do anything involving PAM.

**Comment by ZeroKun265** [comment]
> > Yes I know, which is why I shouldn't be using auto login in the first place and Sddm should just offer a better experience
> 
> I don't know why they don't want to do anything involving PAM.

Had I been smart enough I would have done it, actually I believe some people do but they're not merging the fixes

**Comment by z2rec** [comment]
I commented because some people will see your comment and use your workaround without knowing how it affects them. 

> I don't know why they don't want to do anything involving PAM.

Because they are not payed to do it and there is tons of work that needs to be done since they should change the design which first will involve a lot of discussion on the new design. This will also (most likely) introduce a bunch of new bugs.

In the end it is not a priority at all and we probably won't see this fixed until some hero who *needs* this as part of his workflow, sinks hours of his time into this.

Or maybe it gets fixed once SDDM merges into KDE completely in the future.

**Comment by AkechiShiro** [comment]
I think if people really want this feature/this to be working maybe they should sponsor (opencollective or some other platform to put a bounty on this issue), but I don't know if that is possible or not.

**Comment by 0verEngineer** [comment]
I would absolutely drop some euros if we could direct the money to specific issues

**Comment by ZeroKun265** [comment]
> I think if people really want this feature/this to be working maybe they should sponsor (opencollective or some other platform to put a bounty on this issue), but I don't know if that is possible or not.

Yeah, even small donations could quickly add up, as it seems this isn't just something barely 3 people want.
I'm not in a great financial situation but I could give at least something

**Comment by User8395** [comment]
Has this been updated for the latest version of SDDM?

**Comment by jclsn** [comment]
I would also be open to dropping a dime to fund this. Would be nice to see proper fingerprint authentication. The current implementation is not useful, so I rather just use the password.

**Comment by sophronesis** [comment]
Any updates? 6 years passed already

**Comment by 0verEngineer** [comment]
There are no updates beside the nice Proposal from Enderclem above your comment, i really like the proposal but for SDDM in KDE it is to late: https://blog.davidedmundson.co.uk/blog/a-roadmap-for-a-modern-plasma-login-manager/

**Comment by nphalem** [comment]
From Nate:

> This is happening in https://invent.kde.org/davidedmundson/plasma-login-manager. Once it's ready for a first release, it'll be moved to the Plasma namespace. Closing!

https://invent.kde.org/plasma/plasma-desktop/-/issues/91#note_1188052

It already has been moved to Plasma namespace. And is actively being worked on.

https://invent.kde.org/plasma/plasma-login-manager

**Comment by crizzy9** [comment]
Whats the holdup on this PR?

**Comment by User8395** [comment]
https://github.com/sddm/sddm/pull/1220#issuecomment-1829545241

It's stupid. Instead of having a solution that works fine as others have reported, it's better to wait until this is reworked.
